src/process/io/PathSink.cpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#include "PathSink.h"
+
+#include <experimental/filesystem>
+
+using namespace std;
+using namespace std::experimental::filesystem;
+
+PathSink::PathSink(const path &file, bool forceOverwrite) :
+        Sink(forceOverwrite),
+        file(IOHelper::fullPath(file)),
+        lockHandler(file) {
+}
+
+bool PathSink::fulfillsPremises() {
+    if (!forceOverwrite && exists()) {
+        addErrorMessage(
+                "The file '", file, "' exists and cannot be overwritten. Allow overwriting with -w.");
+        return false;
+    }
+    if (exists() && !canWrite()) {
+        addErrorMessage("The result file '", file, "' exists and cannot be overwritten. Check its file access rights.");
+        return false;
+    }
+//    if (!exists() && access(parentPath().string().c_str(), W_OK) != 0) {
+//        addErrorMessage("The result file cannot be written. Check the access rights of the parent folder.");
+//        return false;
+//    }
+    return true;
+}
+
+bool PathSink::open() {
+    if (!isOpen())
+        this->fStream.open(file, ios_base::out | ios_base::in | ios_base::binary);
```

Comment:
Wondering: Why `ios_base::out` for a sink? I usually hope not to get anything out of my (kitchen) sink :stuck_out_tongue_closed_eyes: